planning.ipynb

```diff
@@ -594,6 +594,104 @@
    "source": [
     "It has now successfully achieved its goal i.e, to build a stack of three blocks."
    ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Have Cake and Eat Cake Too"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "This problem involves the task of eating a cake with an initial condition of having a cake. "
```

Comment:
This is too short of an introduction. As someone who is not familiar with the example, I am confused as to what the problem statement is. Do we simply have a cake and we want to eat it? Or is there something else to it?

Notebook sections are meant to educate people, not simply show the code. You need to be adding more to flesh your sections out.

This goes for your other sections as well. They are just too short. When students read this, they will have questions this notebook will not answer. More detail is needed, sorry.

Reply:
Thank you for your suggestion @MrDupin. I have written this example assuming that the reader has the book, as suggested by @norvig. Also, In the book, this `have_cake_and_eat_cake_too` problem is not explicitly described as the other planning problems like the `air_cargo` or the `spare_tire` problem. This problem is actually used as an example for explaining planning graphs.

Initially, I thought of explaining the planning graph with the help of `have_cake_and_eat_cake_too` in the notebook, but then, that explanation would need a bit lengthy description and I assumed that it would not be desirable to put it in the notebook. If it is advisable, then I would add the planning graph explanation and use the `have_cake_and_eat_cake_too` to describe it.

I've taken reference from the other problems like `air_cargo`, `spare_tire` to see how they are implemented and I've written this accordingly. I understand the need to give a detailed description and I will try to provide some more information. Sorry for the inconvenience. Please let me know what you think @MrDupin.

Reply:
Even though the reader will probably have the book, I find that it is still useful adding complete information about the problem. There is no need to write paragraphs on end, but clearly and cleanly defining the problem is key in making the section useful. Students shouldn't have to go back and forth between the notebook and the book while reading a section just to understand what the problem is about, that is inconvenient. If the students have the minimum needed information on the notebook, they can use it as a supplement to their studying.

I noticed that the Air Cargo and Spare Tire problems also have the same issue. I will shortly open an issue asking for help on those. The Three Block Tower problem on the other hand is a much better gauge of what we are looking for. A short paragraph explaining what is going on will go a long way in helping students.

Reply:
Thanks @MrDupin. I will work on it. Also, for the `have_cake_and_eat_cake_too` problem, I will try to use it as an example for explaining planning graphs. I think it actually needs a bit longer description than what is ideal. I will try to shorten it without compromising on the understanding of the reader.